internal/bminventory/inventory.go

```diff
@@ -2103,60 +2094,65 @@ func (b *bareMetalInventory) updateHostRoles(ctx context.Context, params install
 		}
-		err = b.hostApi.UpdateHostname(ctx, &host, params.ClusterUpdateParams.HostsNames[i].Hostname, db)
+		err = b.hostApi.UpdateHostname(ctx, host, hostName.Hostname, db)
 		if err != nil {
 			log.WithError(err).Errorf("failed to set hostname <%s> host <%s> in cluster <%s>",
-				params.ClusterUpdateParams.HostsNames[i].Hostname, params.ClusterUpdateParams.HostsNames[i].ID,
+				hostName.Hostname, hostName.ID,
 				params.ClusterID)
 			return common.NewApiError(http.StatusConflict, err)
 		}
 	}
 	return nil
 }
 
+func getHost(hostId string, clusterId string, db *gorm.DB, log logrus.FieldLogger) (*models.Host, error){
+	var host models.Host
+	err := db.First(&host, "id = ? and cluster_id = ?", hostId, clusterId).Error
+	if err != nil {
+		log.WithError(err).Errorf("failed to find host <%s> in cluster <%s>", hostId, clusterId)
+		return nil, err
+	}
+	return &host, nil
+}
+
 func (b *bareMetalInventory) updateHostsDiskSelection(ctx context.Context, params installer.UpdateClusterParams, db *gorm.DB, log logrus.FieldLogger) error {
 	for i := range params.ClusterUpdateParams.DisksSelectedConfig {
-		var host models.Host
-		err := db.First(&host, "id = ? and cluster_id = ?",
-			params.ClusterUpdateParams.DisksSelectedConfig[i].ID, params.ClusterID).Error
+		disksConfig := params.ClusterUpdateParams.DisksSelectedConfig[i]
+		hostId := string(disksConfig.ID)
+		host, err := getHost(hostId, string(params.ClusterID), db, log)
+
 		if err != nil {
-			log.WithError(err).Errorf("failed to find host <%s> in cluster <%s>",
-				params.ClusterUpdateParams.DisksSelectedConfig[i].ID, params.ClusterID)
 			return common.NewApiError(http.StatusNotFound, err)
 		}
 
-		var installationDiskPath = ""
-		for _, diskConfig := range params.ClusterUpdateParams.DisksSelectedConfig[i].DisksConfig {
-			if models.DiskRoleInstall == diskConfig.Role {
-				if installationDiskPath != "" {
-					return common.NewApiError(http.StatusConflict,
-						fmt.Errorf("duplicate setting of installation path by the user"))
-				}
-				installationDiskPath = hostutil.FindDiskPathByID(*diskConfig.ID, &host)
-			}
+		disksToInstallOn := funk.Find(disksConfig.DisksConfig, func(diskConfigParams *models.DiskConfigParams) bool {
+			return models.DiskRoleInstall == diskConfigParams.Role
+		}).([]*models.DiskConfigParams)
+
+		installationDiskPath := ""
+
+		if len(disksToInstallOn) > 1{
+			return common.NewApiError(http.StatusConflict, fmt.Errorf("duplicate setting of installation path by the user"))
+		} else if len(disksToInstallOn) == 1 {
+			installationDiskPath = *disksToInstallOn[0].ID
 		}
 
-		log.Infof("Update host %s to install from path %s",
-			params.ClusterUpdateParams.DisksSelectedConfig[i].ID, installationDiskPath)
-		err = b.hostApi.UpdateInstallationDiskPath(ctx, db, &host, installationDiskPath)
+		log.Infof("Update host %s to install from path %s", hostId, installationDiskPath)
+		err = b.hostApi.UpdateInstallationDiskPath(ctx, db, host, installationDiskPath)
```

Comment:
Please modify the description for the installationDiskPath field of the host to explain that its name is misleading and was kept for backwards compatibility purposes, and what it should be called today is "installationDiskID"

Reply:
I documented the change in the swagger.yaml file. change the method name to UpdateInstallationDiskId? (so only the field name will remain the same)

Reply:
> I documented the change in the swagger.yaml file.
I don't see it in the PR diff. I see that you documented the ID field in the inventory disk object, but not in the host "installationDiskPath" field

Reply:
> Change the method name to UpdateInstallationDiskId? (so only the field name will remain the same)
Yeah see my comment below

Reply:
done